### DISCUSSION TITLE
Hot to add astro highlight to react app with monaco editor.  💡 RFC:

### DISCUSSION THREAD

**Discussion by DeeStroKeR** [withastro/language-tools#795] [Support]
### Background & Motivation

I'm working on my own IDE based on Monaco. I want to add support for .astro files in Monaco. I tried using this extension as a source of information, but I can't figure out where I can fetch the information needed to complete this task. Please, help me find a solution.

### Help make it happen!

- [X] I am willing to submit a PR to implement this change.
- [X] I am willing to submit a PR to implement this change, but would need some guidance.
- [X] I am not willing to submit a PR to implement this change.

**Comment by Princesseuh** [comment]
Our language-server is currently not really web friendly, as it depends on Node in many places and doesn't yet load the browser version of the Astro compiler.

**Reply by DeeStroKeR** [reply]
i found npm package (https://github.com/bolinfest/monaco-tm) that converts TM language to monaco format. But highlighting not work well.
![image](https://github.com/withastro/language-tools/assets/24656361/7c10372c-b97c-4802-8641-b78170022d03)
![image](https://github.com/withastro/language-tools/assets/24656361/47b48ca6-6507-4ce6-934d-71954d4716bf)

**Reply by Princesseuh** [reply]
You need the other syntaxes in your app, namely the TypeScript and CSS ones.

**Reply by DeeStroKeR** [reply]
sorry, but I don't understand you. Now I use astro.tmLanguage.json to convert it with a package(monaco tm). Do you mean i need to add ts and css (css.tm.json) to this file?
